src/db.rs

```diff
@@ -30,6 +30,20 @@
 /// Each background worker is signalled with a conditional variable once
 /// there is some work to be done.
 
+// [Review] I am not sure of the use of the `flush_worker`: 
+// it is waiting of `Read` to be empty, so wouldn't it be more
+// straight forward to flush (`fsync`) in the commit_worker.
```

Comment:
The idea is that flushing logs can be done in the background and it should not block anything else really